### DISCUSSION TITLE
Partial dependence plots on Autogluon models

### DISCUSSION THREAD

**Discussion by MridulGoel** [autogluon/autogluon#1354] [Q&A]
Hi I am trying to create Partial dependence plots on independent features present in autogluon model. I took a code from github page:
https://github.com/awslabs/autogluon/tree/master/examples/tabular/interpret

Since it is using kernel explainer to get the shap values, it is taking lot of time to calculate the shap_values for all the train data and also not fitting in my memory size which is 500 gb. Please help me in suggesting someways how i can get the partial dependence plots efficiently. Any guidance would be deeply appreciated.

**Comment by chunyan66** [comment]
Hello, please do you solve this problem? Can you share the answer with me?

**Comment by easyautoml** [comment]
Hi @MridulGoel 

Kernel Explain seem to be worst choice because it's take a lot of time to calculate.
My suggestion is if your model is trained with preset "medium_quality_faster_train". You should load the raw model like "RandomForest" and used Tree Explain instead.
Load raw model is quite easy with "model = self.predictor._trainer.load_model(model_name)"

I developed class name call Explain to calculated SHAP values by use Tree Explain as bellow link. Take a look, if you have any question, lets me know and I will explain more detail to you. 

https://github.com/easyautoml/easyml/blob/6c355850d2879ad31a7d20505919d2ff6e669b94/src/jobs/automl.py#L568

**Reply by easyautoml** [reply]
```
# Step 1. Load raw model. 
model = self.predictor._trainer.load_model(model_name)

# Step 2. Transform data 
df_data_trans = predictor.transform_features(data=df_data, model=model_name)
data = model.preprocess(df_data_trans)

# Step 3. Use Tree Kernel to calcuate shap values
explainer = shap.TreeExplainer(model.model)
shap_values = explainer.shap_values(data)
```

**Reply by MridulGoel** [reply]
Thanks Much. I will give it a try and will let you know if I face any problem

**Reply by rachelledxh** [reply]
@easyautoml I got step1 and step2 running by choosing the catboost model name, however, step 3 is not running and got the error saying that InvalidModelError: Model type not yet supported by TreeExplainer: <class 'NoneType'>. Any thoughts how to solve for it?

**Reply by easyautoml** [reply]
![image](https://github.com/autogluon/autogluon/assets/100541539/e9d5180e-4418-4fec-877c-f7be260a0e61)
You should check is the catboost model none or not before run TreeExplainer

**Reply by rachelledxh** [reply]
@easyautoml Thanks. Under models directory, I see CatBoost_BAG_L1_FULL, and I set 

model_name="CatBoost_BAG_L1_FULL"
catboost_model = predictor._trainer.load_model(model_name)
catboost_model.model is None

and got True
Is that model_name that I used is wrong? If it is wrong, which model name should I use? I tried CatBoost and it is not working.

**Reply by easyautoml** [reply]
There are two points.
**1. :** 
When setting up experiments, choose the preset name "medium_quality_faster_train" to ensure the model remains as simple as possible.
`predictor = task().fit(presets="medium_quality_faster_train")`

To load the model with different presets, such as "best quality", you need to have a deep understanding of what Autogluon does with that preset. Before moving on to another mode, ensure you understand its implications. While I can't delve into all the details, it's crucial to comprehend the differences between these presets, especially if you plan to load the raw model and use libraries like SHAP for further analysis.

**2.:**
For the model name used to load model, you should get from predictor.leaderboard().
